<a href="https://colab.research.google.com/github/Dr-Mamiz/Projects-Data-Analysis/blob/main/05_%232_Project_May_2022.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Import packages
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime

In [ ]:
# Import dataset
df= pd.read_csv('/content/conlog.csv.bz2')

In [4]:
#Columns in our dataset-take a look
df.head(3)

cu_id                          time   node   operation
0  5447683  2021-02-11 08:29:34.653+0000  COMM2  DISCONNECT
1  5447683  2021-02-11 08:17:45.353+0000  COMM2     CONNECT
2  5447683  2021-02-11 08:17:45.351+0000  COMM2  DISCONNECT

In [5]:
#Nb of raws in our df
df.count()

cu_id        33984140
time         33984140
node         33984140
operation    33984140
dtype: int64

**In our dataset almost 34 million raws**

In [7]:
# Dataset SHAPE
df.shape

(33984140, 4)

**We've 135.936.560 values enteries**

In [ ]:
# Checking if we have any weird values
df.describe()

In [8]:
# Basic information about our dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33984140 entries, 0 to 33984139
Data columns (total 4 columns):
 #   Column     Dtype 
---  ------     ----- 
 0   cu_id      int64 
 1   time       object
 2   node       object
 3   operation  object
dtypes: int64(1), object(3)
memory usage: 1.0+ GB


In [35]:
#All unique CUs+ CU's NB
df.cu_id.value_counts()

266896305    661078
203050455    408603
804252384    397762
103747987    329552
203053807    316793
              ...  
190012900         2
190033600         2
190031500         2
5376929           2
190006900         2
Name: cu_id, Length: 6473, dtype: int64

**We've 6473 differente CUs**

In [55]:
# NB of CUs with enteries *READINGS* < 40
(df.cu_id.value_counts()<40).value_counts()

False    5108
True     1365
Name: cu_id, dtype: int64

**We've 1365 CU's with readings < 40 and 5108 CUs with more than 40**

In [ ]:
#Split the column time into Date and Time
df[['Date','Time']] = df['time'].str.split(" ",expand=True)

In [ ]:
#Checking that the df
df.head(2)

In [ ]:
 #(THEN) Extract month 
df['Month']=df['time'].apply(month)
df.head(3)